core/dogu_v1.go

```diff
@@ -12,6 +13,7 @@ import (
 type DoguV1 struct {
 	Name                 string
 	Version              string
+	PublishedAt          time.Time
```

Comment:
Do you want to operate on a zero value for time.Time? If so, marshaling will create json with initial values as you can see in your tests above. Another option is to use PublishedAt as a pointer together with the json tag:

`PublishedAt          time.Time \`json:",omitempty"\``  this way PublishedAt will not be written when it is empty.  

https://go.dev/play/p/6nqE7CmRJL2

Reply:
We don't have a particular opinion on this and would change it if you prefer.
The field will actually always be set and imho a zero value can/should not occur.